### PR TITLE
5133 work done; 3780 post modifications; 3217 dymic data change

### PR DESCRIPTION
**Pull request by SahitiVemulapalli**
### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Review and publish page updated
- [ ] UI changes look good on mobile

### Developer self-QA run statement

- [ ] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts [hunk 1]
```diff
@@ -13,24 +13,19 @@ const excelJS= require('exceljs');
 export const CA_GET_REVIEW_RANKED_SUPPLIERS = async (req: express.Request, res: express.Response) => {
   const { SESSION_ID } = req.cookies; //jwt
   const { projectId, releatedContent, isError, errorText, eventId, currentEvent } = req.session;
+  const { assessmentId } = currentEvent;
```
Comment (csk95) [inline src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts:16]: Are you using assessmentId in this page ?

Comment (SahitiVemulapalli) [reply src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts:16]: yes i need to use to get the related data

File: src/main/features/requirements/controller/ca-whereWorkDone.ts [hunk 2]
```diff
@@ -7,22 +7,37 @@ import { LoggTracer } from '../../../common/logtracer/tracer';
 
 export const CA_GET_WHERE_WORK_DONE = async (req: express.Request, res: express.Response) => {
   const { SESSION_ID } = req.cookies; //jwt
-  const { projectId, releatedContent, isError, errorText, dimensions } = req.session;
+  const { projectId, releatedContent, isError, errorText, dimensions,currentEvent } = req.session;
```
Comment (csk95) [inline src/main/features/requirements/controller/ca-whereWorkDone.ts:10]: Is this currentEvent declaration really required?

Comment (SahitiVemulapalli) [reply src/main/features/requirements/controller/ca-whereWorkDone.ts:10]: yes to get the assessmentID

File: src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts (hunk 1 above)
Comment (csk95) [reply src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts:16]: I can't see use of assessmentId in this page

Comment (SahitiVemulapalli) [reply src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts:16]: check the line number 83

**Review by BalwinderSinghCts — CHANGES_REQUESTED** [review]
src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts
Line number 25
`dimensionRequirements.filter(dimension => dimension["dimension-id"] === 1)[0]`

null check are not implemented if are you using indexing then we need to add checks for null

File: src/main/features/requirements/controller/ca-whereWorkDone.ts (hunk 2 above)
Comment (csk95) [reply src/main/features/requirements/controller/ca-whereWorkDone.ts:10]: to get assessmentID, you already use as "req.session.currentEvent.assessmentId" in  line number 15.

File: src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts (hunk 1 above)
Comment (csk95) [reply src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts:16]: That's commented code

Comment (SahitiVemulapalli) [reply src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts:16]: the commented code will be used when all the pages wrk

Comment (csk95) [reply src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts:16]: Then you should also comment this code whenever it required then we will uncomment this code

**Review by BalwinderSinghCts — CHANGES_REQUESTED** [review]
`src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts`
line no 36 
1) toLower or toUpper function are not used 
2) static data remove 'Security Clearance'  and add into common file and use as enums 
if(dimensionRequirements.filter(dimension => dimension.name === 'Security Clearance')[0].requirements.length>0)

File: src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts (hunk 1 above)
Comment (SahitiVemulapalli) [reply src/main/features/requirements/controller/ca-reviewRankedSuppliers.ts:16]: does that cause any issue now?

**Comment by SahitiVemulapalli** [comment]
> src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts Line number 25 `dimensionRequirements.filter(dimension => dimension["dimension-id"] === 1)[0]`
> 
> null check are not implemented if are you using indexing then we need to add checks for null

the check is done in the prvs if statements.

File: src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts [hunk 3]
```diff
@@ -20,17 +20,21 @@ export const CA_GET_CHOOSE_SECURITY_REQUIREMENTS = async (req: express.Request,
...
     if (dimensionRequirements.length > 0) {
-        if(dimensionRequirements.filter(dimension => dimension["dimension-id"] === 7).length>0)
+        if(dimensionRequirements.filter(dimension => dimension["dimension-id"] === 1).length>0)
       {
-        totalQuantityca=dimensionRequirements.filter(x=>x["dimension-id"])[0].requirements.map(a => a.weighting).reduce(function(a, b)
-      {
-         return a + b;
-      });
-      req.session.totalQuantityca=totalQuantityca;
+        if(dimensionRequirements.filter(dimension => dimension["dimension-id"] === 1)[0].requirements.length>0)
```
Comment (BalwinderSinghCts) [inline src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts:25]: 1.  Null Checks are not implemented for indexing 
2.  null checks are not added for this dimension["dimension-id"]

File: src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts [hunk 4]
```diff
@@ -20,17 +20,21 @@ export const CA_GET_CHOOSE_SECURITY_REQUIREMENTS = async (req: express.Request,
...
+        {
+          totalQuantityca=dimensionRequirements.filter(x=>x["dimension-id"])[0].requirements.map(a => a.weighting).reduce(function(a, b)
+          {
+             return a + b;
+          });
+          req.session.totalQuantityca=totalQuantityca;
+        }      
     }
      if(dimensionRequirements.filter(dimension => dimension.name === 'Security Clearance').length>0)
```
Comment (BalwinderSinghCts) [inline src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts:34]: 1. Add null checks for Array 
2. Put static data in different file **Security Clearance** use as ENUMS and use ToLower or Toupper function for this kind of code

File: src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts [hunk 5]
```diff
@@ -20,17 +20,21 @@ export const CA_GET_CHOOSE_SECURITY_REQUIREMENTS = async (req: express.Request,
...
       {
-
+        if(dimensionRequirements.filter(dimension => dimension.name === 'Security Clearance')[0].requirements.length>0)
```
Comment (BalwinderSinghCts) [inline src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts:36]: 1. Index of null is not checks are implemented `dimensionRequirements.filter(dimension => dimension.name === 'Security Clearance')[0]`

**Comment by SahitiVemulapalli** [comment]
> `src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts` line no 36
> 
> 1. toLower or toUpper function are not used
> 2. static data remove 'Security Clearance'  and add into common file and use as enums
>    if(dimensionRequirements.filter(dimension => dimension.name === 'Security Clearance')[0].requirements.length>0)

It is not static Data it is coming from BACKEND API.

File: src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts (hunk 3 above)
Comment (SahitiVemulapalli) [reply src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts:25]: the check was made in before if statement 
 if(dimensionRequirements.filter(dimension => dimension["dimension-id"] === 1).length>0)

File: src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts (hunk 5 above)
Comment (SahitiVemulapalli) [reply src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts:36]: the check is made in  if(dimensionRequirements.filter(dimension => dimension.name === 'Security Clearance').length>0)

File: src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts (hunk 4 above)
Comment (SahitiVemulapalli) [reply src/main/features/requirements/controller/ca-ChooseSecurityRequirements.ts:34]: "Security Clearance" is  NOT STATIC DATA it is from BE API

**Review by BalwinderSinghCts — APPROVED** [review]

**Review by dhivya-cognizant — APPROVED** [review]

**Pull request merged**